app/decorators/table_sorters/hearings_sorter.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module TableSorters
+  class HearingsSorter
+    def initialize(hearings, sort_order)
+      @hearings = hearings
+      @sort_order = sort_order
+    end
+
+    def sorted_hearings
+      order_by_asc_or_desc(@hearings.sort_by { |h| h.hearing_days.map(&:to_datetime) })
+    end
+
+    def sorted_hearing(hearing)
+      if @sort_order.eql?('date_desc')
+        hearing.hearing_days.map(&:to_datetime).sort.reverse
+      else
+        hearing.hearing_days.map(&:to_datetime).sort
+      end
+    end
+
+    private
+
+    def order_by_asc_or_desc(sorted_hearings)
+      sorted_hearings.reverse! if @sort_order&.include? 'desc'
```

Comment:
1. This will modify `sorted_hearings`. Would it work to do this?

```suggestion
      return sorted_hearings.reverse if @sort_order&.include? 'desc'
```

2. Is there a reason not to have `@sort_order == 'desc'`?
3. I'd suggest changing the argument so that it doesn't mask the `sorted_hearings` method. I just found it a little confusing.

Reply:
Regarding (2), I have just seen that the order can be `date_desc`.

Reply:
@jrmhaig, thanks for the suggestions, I have made the change you suggested in 1. and renamed the argument for order_by_asc_or_desc to arr.